File: codewar/6-kyu/countString.js
// https://www.codewars.com/kata/52efefcbcdf57161d4000091/train/javascript

/**
 * 
 * @param {string} string 
 * @returns 
 */
function count(string) {
  const res = {};
  // string.split("").forEach((char) => {
  //   res[char] = res[char] ? res[char] + 1 : 1;
  // })
  for (let i = 0; i < string.length; i++) {
    const char = string[i];
    res[char] = res[char] ? res[char] + 1 : 1;
  }
  return res;
}

console.log(count("aba"));

File: codewar/count_string.py
# https://www.codewars.com/kata/52efefcbcdf57161d4000091/train/python

def count(string: str) -> dict:
    res = {}
    for char in string:
        res[char] = string.count(char)
    return res

print(count("aba"))



def count2(s: str) -> dict:
    res = {}
    for char in s:
        if char in res:
            res[char] = res[char] + 1
        else:
            res[char] = 1
    return res


from collections import Counter

def count3(s: str) -> dict:
    return dict(Counter(s))

print(count3("aba"))
